src/loginRequest.js

```diff
@@ -0,0 +1,24 @@
+export const login = async (email, password) => {
+  try {
+    const response = await fetch("/auth/login", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        email,
+        password,
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error("Login failed");
+    }
+
+    const data = await response.json();
+    return { loginSuccess: true, data };
+  } catch (error) {
+    console.error("Login Error:", error);
+    return { loginSuccess: false, error: error.message };
+  }
```

Comment:
So you're using a concept here where you're returning a boolean value for if it succeeded along with the return value and thats totally valid. The programming languages Go and Rust do something similar where any function that can have an error will return both an error and a value

Here is an example of how errors are handled in Go. Functions can return both a value and an error and only one of them should actually hold some info or data.
```go
value, error = someFunctionThatSometimesErrors()
if (error != none) {
  // handle the error here
}

// otherwise this means there was no error so you can assume there is a value
doThing(value)

```

So you could also do this in javascript where you can modify every function that can return an error so that it creates an object that contains the success boolean and has an actual return value like you did in this file. 
But generally because of how javascript is designed to have errors that can be thrown, pretty much everybody just uses throwing errors as the equivalent of returning loginSuccess: false. So I think we should stick with the convention of throwing errors in our project as well. 

So basically you can just let this function throw an error if something goes wrong and then in your component, just use try catch instead of `if (loginSuccess) { } else {}`

A function should return one particular thing and if it cant do that for whatever reason then it should throw an error.


Let me know if any of this doesn't make sense or if you want more examples!